digdag-guice-rs-server-undertow/src/main/java/io/digdag/guice/rs/server/undertow/UndertowServer.java

```diff
@@ -86,6 +96,55 @@ ServerBootstrap getServerBootstrap()
         }
     }
 
+    private static AttachmentKey<String> LISTEN_ADDRESS_NAME_ATTACHMENT = AttachmentKey.create(String.class);
+
+    static class SetListenAddressNameHandler
+            implements HttpHandler
+    {
+        private final HttpHandler next;
+        private final String name;
+
+        public SetListenAddressNameHandler(HttpHandler next, String name)
+        {
+            this.next = next;
+            this.name = name;
+        }
+
+        @Override
+        public void handleRequest(HttpServerExchange exchange)
+                throws Exception
+        {
+            exchange.putAttachment(LISTEN_ADDRESS_NAME_ATTACHMENT, name);
+            next.handleRequest(exchange);
+        }
+    }
+
+    static class SetListenAddressNameServletFilter
+            implements Filter
+    {
+        @Override
+        public void init(FilterConfig filterConfig)
+        { }
+
+        @Override
+        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+                throws IOException, ServletException
+        {
+            if (request instanceof HttpServletRequestImpl) {
+                HttpServerExchange exchange = ((HttpServletRequestImpl) request).getExchange();
+                String name = exchange.getAttachment(LISTEN_ADDRESS_NAME_ATTACHMENT);
+                if (name != null) {
+                    request.setAttribute(GuiceRsServerRuntimeInfo.LISTEN_ADDRESS_NAME_ATTRIBUTE, name);
+                }
+                chain.doFilter(request, response);
```

Comment:
According to JavaDoc, here needs to call `chain.doFilter` if here doesn't want to block request processing.

https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/Filter.html#doFilter(javax.servlet.ServletRequest, javax.servlet.ServletResponse, javax.servlet.FilterChain)